### PR TITLE
New features

### PR DESCRIPTION
**Pull request by soumyart**
**New Features**

1. Split Dock.
2. Resize Panel (In undocked state).
3. Highlight drop target.
4. Ability to Replace underlying components of AreaDock, Dock and Panels.

**Updates**
1. Updated Panel z-index to start from 2001.
2. Updated ReadMe.
3. Used cross-env to set the development port.

### CODE REVIEW

**Comment by amh4r** [comment]
Thanks for the PR, @soumyart! The new features seem to be working properly for the most part.

Here are some changes/questions:

1. When docking a panel into the right area dock, there's a 2px gap. I'm not sure why this is happening. Maybe it has something to do with the `border` that shows where a panel will dock. If that's causing the issue, maybe we can use `outline` instead of `border`.

image

2. Why switch to `react-rnd` from `react-draggable`? I don't have anything against the change; I'm just curious why `react-rnd` was needed.

3. The new `components` prop in `DragAndDock.Area.Dock` and `DragAndDock.Panel` is essentially an object of render prop functions. Could you please move these functions into their own prop methods? For example, `components. DockContainer()` would become `renderDockContainer()`. This would be keeping with the pattern established by the `renderTitleBar()` prop in `DragAndDock.Panel`.

File: src/Dock/index.js [hunk 1]
```diff
@@ -75,19 +107,22 @@ function Dock(props) {
...
+      y:
+        dockRect.y +
+        window.scrollY +
+        (tabLocation === 'bottom' ? dockRect.height - dock.panelTabsHeight + 2 : 0),
     };
   })();
 
-  const width = dockRect ? dockRect.width : null;
-
+  const width = dockRect ? dockRect.width - 2 : null;
```
Comment (amh4r) [inline src/Dock/index.js:117]: Why was this 2px offset needed?

**Comment by soumyart** [comment]
1. Yup I think outline makes sense. 2px was subtracted as a compensation for border. Will revert that to use outline.

2. React-rnd has resize option along with drag. For now you can only resize non docked panels. Resizing the panel while it is docked needs some additional code so I have disabled it for now. I think having resizable panel will be a nice feature.

3. In the prop type I have mentioned components is an object which contains render functions but I meant them to be react components instead of plain func. So when user wants to replace a component they can directly assign custom react component to one of the components key and props will be passed to those custom components as documented in ReadMe.
Instead of individually exposing function if we can wrap them through a single object, I think It will be a cleaner approach to replace an underlying component in Panel/Dock. 
I got this idea from https://react-select.com/components.

But still we can go with your approach also.

File: src/Dock/index.js (hunk 1 above)
Comment (soumyart) [reply src/Dock/index.js:117]: 2px was subtracted as a compensation for border. Will revert that in favour of outline.

**Comment by amh4r** [comment]
Yea, since we have multiple component overrides it probably better that we put them all in the same prop, rather than having multiple render props.

I changed the base branch to `release-0.3`. Since removing the `renderTitleBar` prop will be a breaking change, we'll need to bump the version to 0.3.

**Pull request merged**